### PR TITLE
Speed up GSI analysis jobs in CI testing

### LINKED ISSUE #3114: Speed up GSI run in CI tests
### What new functionality do you need?

In the workflow CI cycling tests, GSI runs with the 'default' number of iterations. These tests are not scientific, so there is no need to run this many variational iterations. We can reduce this in 'testing' mode to speed up analysis jobs.

### What are the requirements for the new functionality?

Add in a flag that indicates the CI test is running so make the configuration 'faster/simpler'.

### Acceptance Criteria

- [ ] cycled tests pass
- [ ] some sort of reduction in analysis job walltime

### Suggest a solution (optional)

_No response_

### PR DESCRIPTION
**Pull request by CoryMartin-NOAA**
# Description
This PR adds `DO_TEST_MODE`, which can be used for other things in the future but for now sets the GSI to run just 5 iterations per outer loop to reduce runtime for CI testing.

  Resolves #3114 

# Type of change
- [ ] Bug fix (fixes something broken)
- [ ] New feature (adds functionality)
- [x] Maintenance (code refactor, clean-up, new CI test, etc.)

# Change characteristics
- Is this a breaking change (a change in existing functionality)? NO
- Does this change require a documentation update? MAYBE
- Does this change require an update to any of the following submodules? NO

# How has this been tested?
- Clone and build on Hera
- Cycled test on Hera

# Checklist
- [x] Any dependent changes have been merged and published
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have documented my code, including function, input, and output descriptions
- [x] My changes generate no new warnings
- [x] New and existing tests pass with my changes
- [x] This change is covered by an existing CI test or a new one has been added
- [x] Any new scripts have been added to the .github/CODEOWNERS file with owners
- [x] I have made corresponding changes to the system documentation if necessary

### CODE REVIEW

**Review by RussTreadon-NOAA — APPROVED** [review]
Straightforward PR.  Approve ... but did not test.

File: ci/cases/yamls/atmaerosnowDA_defaults_ci.yaml
Comment (DavidNew-NOAA) [inline ci/cases/yamls/atmaerosnowDA_defaults_ci.yaml:6]: Since I plan on eventually adding ```DO_JEDI_TESTING: "YES"``` here as well once we have test references for JEDI aero and snow DA, would it make sense to change this to ```DO_GSI_TESTING``` or something like that?

Comment (CoryMartin-NOAA) [reply ci/cases/yamls/atmaerosnowDA_defaults_ci.yaml:6]: Good question, are there other potential uses of the testing flag beyond DA? Asking g-w code maintainers for their input.

Comment (DavidNew-NOAA) [reply ci/cases/yamls/atmaerosnowDA_defaults_ci.yaml:6]: Alternatively, ```DO_TEST_MODE``` could turn on testing mode for both GSI and JEDI

Comment (DavidHuber-NOAA) [reply ci/cases/yamls/atmaerosnowDA_defaults_ci.yaml:6]: I think I would prefer a single flag to indicate testing mode.  It's possible we could find a good use for a test mode with the forecast as well (perhaps shortening forecasts to 48 hours).  We could then run with `DO_TEST_MODE=NO` for weekly tests to capture the full range of the analyses and forecasts.

Comment (DavidNew-NOAA) [reply ci/cases/yamls/atmaerosnowDA_defaults_ci.yaml:6]: I agree. I'll update GW PR [#3120](https://github.com/NOAA-EMC/global-workflow/pull/3120) to use this flag as well.

**Review by DavidNew-NOAA — APPROVED** [review]

**Review by DavidHuber-NOAA — APPROVED** [review]

**Comment by JessicaMeixner-NOAA** [comment]
@WalterKolczynski-NOAA  I was pointed to this failure offline because it was about a wave IC issue: 

`SError: unable to copy /scratch1/NCEPDEV/global/glopara/data/ICSDIR/C48mx500/20240610/gfs.20210323/06/model/wave/restart/20210323.120000.restart.ww3 to /scratch1/NCEPDEV/global/CI/3115/RUNTESTS/COMROOT/C48_S2SW_0bb9078e/gfs.20210323/06//model/wave/restart/20210323.120000.restart.ww3 `

I'm concerned that changes in staging of wave ICs has messed up existing regression tests in prep for https://github.com/NOAA-EMC/global-workflow/pull/3112 
removing the grid id perhaps is causing an issue because we have many wave gridids that are used in various places and what was copied to ww3?   And where?   I'll continue this conversation in 3112.  I don't think we want to break other tests before 3112 is merged do we?   And perahps we need to carefully go through and make sure ICs are properly placed for various wave grids for various CI tests.  I'm unsure about this now.

**Comment by WalterKolczynski-NOAA** [comment]
> @WalterKolczynski-NOAA I was pointed to this failure offline because it was about a wave IC issue:
> 
> `SError: unable to copy /scratch1/NCEPDEV/global/glopara/data/ICSDIR/C48mx500/20240610/gfs.20210323/06/model/wave/restart/20210323.120000.restart.ww3 to /scratch1/NCEPDEV/global/CI/3115/RUNTESTS/COMROOT/C48_S2SW_0bb9078e/gfs.20210323/06//model/wave/restart/20210323.120000.restart.ww3 `
> 
> I'm concerned that changes in staging of wave ICs has messed up existing regression tests in prep for #3112 removing the grid id perhaps is causing an issue because we have many wave gridids that are used in various places and what was copied to ww3? And where? I'll continue this conversation in 3112. I don't think we want to break other tests before 3112 is merged do we? And perahps we need to carefully go through and make sure ICs are properly placed for various wave grids for various CI tests. I'm unsure about this now.

A few symlinks got broken somehow. Breaking old tests was not intentional. The links are restored now.

**Comment by DavidHuber-NOAA** [comment]
Relaunching tests on Hera.

**Review by WalterKolczynski-NOAA — APPROVED** [review]

**Pull request merged**
